dev/PagerControl/PagerControl.cpp

```diff
@@ -0,0 +1,773 @@
+        }
+    }
+    if (PreviousButtonVisibility() == winrt::PagerControlButtonVisibility::HiddenOnEdge)
+    {
+        if (selectedPageIndex != 0)
+        {
+            winrt::VisualStateManager::GoToState(*this, c_previousPageButtonVisibleVisualState, false);
+        }
+        else
+        {
+            winrt::VisualStateManager::GoToState(*this, c_previousPageButtonHiddenVisualState, false);
+        }
+    }
+    if (NextButtonVisibility() == winrt::PagerControlButtonVisibility::HiddenOnEdge)
+    {
+        if (selectedPageIndex != numberOfPages - 1)
+        {
+            winrt::VisualStateManager::GoToState(*this, c_nextPageButtonVisibleVisualState, false);
+        }
+        else
+        {
+            winrt::VisualStateManager::GoToState(*this, c_nextPageButtonHiddenVisualState, false);
+        }
+    }
+    if (LastButtonVisibility() == winrt::PagerControlButtonVisibility::HiddenOnEdge)
+    {
+        if (selectedPageIndex != numberOfPages - 1)
+        {
+            winrt::VisualStateManager::GoToState(*this, c_lastPageButtonVisibleVisualState, false);
+        }
+        else
+        {
+            winrt::VisualStateManager::GoToState(*this, c_lastPageButtonHiddenVisualState, false);
+        }
+    }
+}
+
+/* NumberPanel logic */
+void PagerControl::UpdateNumberPanel(const int numberOfPages)
+{
+    if (numberOfPages < 0)
+    {
+        UpdateNumberOfPanelCollectionInfiniteItems();
+    }
+    else if (numberOfPages < 7)
+    {
+        UpdateNumberPanelCollectionAllItems(numberOfPages);
+    }
+    else
+    {
+        const auto selectedIndex = SelectedPageIndex();
+        // Idea: Choose correct "template" based on SelectedPageIndex (x) and NumberOfPages (n)
+        // 1 2 3 4 5 6 ... n <-- Items
+        if (selectedIndex < 4)
+        {
+            // First two items selected, create following pattern:
+            // 1 2 3 4 5... n
+            UpdateNumberPanelCollectionStartWithEllipsis(numberOfPages, selectedIndex);
+        }
+        else if (selectedIndex >= numberOfPages - 4)
+        {
+            // Last two items selected, create following pattern:
+            //1 [...] n-4 n-3 n-2 n-1 n
+            UpdateNumberPanelCollectionEndWithEllipsis(numberOfPages, selectedIndex);
+        }
+        else
+        {
+            // Neither start or end, so lets do this:
+            // 1 [...] x-2 x-1 x x+1 x+2 [...] n
+            // where x-2 > 1 and x+2 < n
+            UpdateNumberPanelCollectionCenterWithEllipsis(numberOfPages, selectedIndex);
+        }
+    }
+}
+
+void PagerControl::UpdateNumberOfPanelCollectionInfiniteItems()
+{
+    const int selectedIndex = SelectedPageIndex();
+
+    m_numberPanelElements.Clear();
+    if (selectedIndex < 3)
+    {
+        AppendButtonToNumberPanelList(1, 0);
+        AppendButtonToNumberPanelList(2, 0);
+        AppendButtonToNumberPanelList(3, 0);
+        AppendButtonToNumberPanelList(4, 0);
+        AppendButtonToNumberPanelList(5, 0);
```

Comment:
>AppendButtonToNumberPanelList [](start = 8, length = 29)

don't we normally show 7 elements in the panel?

Reply:
I couldn't find anything regarding this in the spec. Should I update this to be 7? @MarissaMatt FYI